### PR TITLE
Extended settings

### PR DESCRIPTION
**Pull request by jstorm31**
Added settings for scaffolding of components:

- module dependencies (whether to load from `dependencies.js` file or not)
- fela hook instead of `connect` HOC
- typescript version of all templates

Also fixed audit vulnerabilities.

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -47,16 +57,62 @@ export default class Model {
...
+  createIndexFile(componentName: string, path: string, fela: boolean) {
+    const configuration = vscode.workspace.getConfiguration(config.namespace);
+    const name = `index.${configuration.typescript ? "t" : "j"}s`;
+
+    let content;
+    if (!fela || (fela && configuration.felaHooks)) {
+      content = templatesJavascript.index(componentName);
+    } else {
+      content = templatesJavascript.indexFela(componentName);
+    }
```
Comment (horaklukas) [inline src/model.ts:111]:
```suggestion
    if (fela && !configuration.felaHooks) {
        content = templatesJavascript.indexFela(componentName);
    } else {
        content = templatesJavascript.index(componentName);
    }
```

File: src/template/component.ts [hunk 2]
```diff
@@ -24,7 +30,41 @@ ${name}.propTypes = {
 export default ${name};
 `;
 
-export const component = name => `import { React, PropTypes } from '../../dependencies';
+export const felaHookComponent = (name: string, dependencies: boolean) => `${
+  dependencies
+    ? "import { React, PropTypes } from '../../dependencies';"
+    : "import React from 'react';\nimport PropTypes from 'prop-types';"
+}
+
```
Comment (horaklukas) [inline src/template/component.ts:38]: Maybe add draft of hook import `import useFelaEnhanced from ''`

Comment (horaklukas) [reply src/template/component.ts:38]: Just noticed it's already present in typescript templates so pls add it here too 🙏  `import { useFelaEnhanced } from 'hooks'`

File: src/template/componentsTypescrip.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+export const felaComponent = (name: string, dependencies: boolean) => `${
+    dependencies
+      ? "import { React,  FunctionComponent } from '../../dependencies';"
+      : "import React, { FunctionComponent } from 'react';"
+  }
+
+type ${name}Props = {
+    styles: object
+}
```
Comment (horaklukas) [inline src/template/componentsTypescrip.ts:5]: I personally prefer 

```ts
interface ${name}Props = {
   styles: object;
}

Comment (horaklukas) [reply src/template/componentsTypescrip.ts:5]: you can also use `FelaWithStylesProps` here 

```ts

export interface ${name}OwnProps {}

type ${name}Props = ${name}OwnProps &  FelaWithStylesProps<${name}OwnProps, {}> 
```

**Review by horaklukas — APPROVED** [review]
Really great job 👏 just what I need, will use immediately when it's released

File: src/template/component.ts (hunk 2 above)
Comment (jstorm31) [reply src/template/component.ts:38]: I forgot to add it there 👍

File: src/template/componentsTypescrip.ts (hunk 3 above)
Comment (jstorm31) [reply src/template/componentsTypescrip.ts:5]: Good idea, I changed it 👍

**Pull request merged**